### PR TITLE
feat($cli): Add support for --no-scripts

### PR DESCRIPTION
**Pull request by erikfox**
#143

**What**: Running a script will by default output the command text of the script to stdout, passing
--no-scripts will hide the command text.

**Why**: Some script commands are long and clutter console output when logged.

**How**: Added `{ scripts: ... }` to options. Defaults `true`, pass `--no-scripts` sets to `false`.

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -85,6 +85,19 @@ test('options: logLevel sets the log level', () => {
...
+test('options: scripts prevents command text when false', () => {
+  const {runPackageScript} = setup()
+  const scriptConfig = {test: {script: 'echo test'}}
+  const options = {scripts: false}
+  return runPackageScript({
+    scriptConfig,
+    scripts: ['test'],
+    options,
+  }).then(() => {
+    // expect(stdout).toEqual(...)
```
Comment (erikfox) [inline src/index.test.js]: @kentcdodds Could you advise on how you would capture stdout to test this? Currently no precedent in tests (that I've seen).

**Comment by kentcdodds** [comment]
This is great! Thanks for working on it! I'll try to get to it as soon as I can (at a conference right now).

**Comment by tdeschryver** [comment]
With looking at the code, I think we could use the `infoSpy` from `setup()`.
```
const {runPackageScript, infoSpy} = setup()
// and in the test
expect(infoSpy.calledWith(...)) // or something familiar
```

File: src/index.test.js (hunk 1 above)
Comment (tdeschryver) [inline src/index.test.js]: With looking at the code, I think we could use the `infoSpy` from `setup()`.
```
const {runPackageScript, infoSpy} = setup()
// and in the test
expect(infoSpy.calledWith(...)) // or something familiar
```

Comment (erikfox) [inline src/index.test.js]: Thanks @tdeschryver will try this tonight.

File: src/index.test.js [hunk 2]
```diff
@@ -85,6 +88,21 @@ test('options: logLevel sets the log level', () => {
...
+test('options: scripts prevents command text when false', () => {
+  const {runPackageScript, infoSpy} = setup()
+  const scriptConfig = {test: {script: 'echo test'}}
+  const options = {scripts: false}
+  return runPackageScript({
+    scriptConfig,
+    scripts: ['test'],
+    options,
+  }).then(() => {
+    expect(infoSpy.firstCall.args[0]).toMatch(
```
Comment (erikfox) [inline src/index.test.js]: @tdeschryver @kentcdodds Let me know if this is acceptable for a test case.

File: src/index.test.js [hunk 3]
```diff
@@ -85,6 +88,36 @@ test('options: logLevel sets the log level', () => {
...
+test('options: scripts logs command text when false', () => {
+  const {runPackageScript, infoSpy} = setup()
+  const scriptConfig = {test: {script: 'echo test'}}
+  const options = {scripts: true}
+  return runPackageScript({
+    scriptConfig,
+    scripts: ['test'],
+    options,
+  }).then(() => {
+    expect(infoSpy.firstCall.args[0]).toMatch(
```
Comment (kentcdodds) [inline src/index.test.js]: Why don't we do `toMatchSnapshot` instead? Will be easier to update in the future.

File: src/index.test.js [hunk 4]
```diff
@@ -85,6 +88,36 @@ test('options: logLevel sets the log level', () => {
...
+  const {runPackageScript, infoSpy} = setup()
+  const scriptConfig = {test: {script: 'echo test'}}
+  const options = {scripts: false}
+  return runPackageScript({
+    scriptConfig,
+    scripts: ['test'],
+    options,
+  }).then(() => {
+    expect(infoSpy.firstCall.args[0]).toMatch(
+      '\u001b[90mnps is executing\u001b[39m `\u001b[1mtest\u001b[22m`',
```
Comment (kentcdodds) [inline src/index.test.js]: Let's do `toMatchSnapshot` here as well.

**Comment by erikfox** [comment]
@kentcdodds Done!
- [x] Document in `README.md`
- [x] `.toMatchSnapshot()`

**Comment by kentcdodds** [comment]
Thank you so much!

**Comment by RichDonnellan** [comment]
Awesome! Can't wait to use it.

Next time I'll try my hand at my own feature request. 👍

**Pull request merged**